ethers-core/src/utils/genesis.rs

```diff
@@ -31,13 +35,33 @@ pub struct Genesis {
     pub difficulty: U256,
 
     /// The genesis header mix hash.
+    #[serde(default)]
     pub mix_hash: H256,
 
     /// The genesis header coinbase address.
+    #[serde(default)]
     pub coinbase: Address,
 
     /// The initial state of the genesis block.
     pub alloc: HashMap<Address, GenesisAccount>,
+
+    // The following fields are only included for tests, and should not be used in real genesis
+    // blocks.
+    /// The block number
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    pub number: Option<U64>,
+
+    /// The block gas gasUsed
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    pub gas_used: Option<U64>,
+
+    /// The block parent hash
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    pub parent_hash: Option<H256>,
+
+    /// The base fee
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    pub base_fee_per_gas: Option<U256>,
```

Comment:
I would consider putting this in a `hive: HiveParams` w/ `serde(flatten)` but we can leave for follow-up, we don't use the genesis type in other places either way

Reply:
ah, yeah that makes sense